lib/irc/Scheduler.js

```diff
@@ -0,0 +1,107 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var promiseutil = require("../promiseutil");
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+
+function ScheduledQueue (server) {
+    this._queue = [];
+    this._processing = null;
+    this._server = server;
+
+    setInterval(this._consume.bind(this), server.getReconnectInterval());
+}
+
+ScheduledQueue.prototype._consume = Promise.coroutine(function*() {
+    if (this._processing) {
+        return;
+    }
+    this._processing = this._queue.shift();
+    if (!this._processing) {
+        return;
+    }
+    try {
+        let thing = this._procFn(this._processing);
+
+        let result = yield thing;
+        log.info(`Resolving scheduled promise for ${this._server.domain}`);
+        this._processing.defer.resolve(result);
+    }
+    catch (err) {
+        log.info(`Rejecting scheduled promise for ${this._server.domain} (${err.message})`);
+        this._processing.defer.reject(err);
+    }
+    finally {
+        this._processing = null;
+    }
+});
+
+ScheduledQueue.prototype._procFn = function (item) {
+    return item.fn();
+}
+
+ScheduledQueue.prototype.enqueue = function(item) {
```

Comment:
Please take care with your spacing. This line is `function(item) {` whereas `:40` is `function (item) {`. Use the same form we use elsewhere please (no extra space).